### ISSUE TITLE
[Day 3] Dense 신경망으로 식물 병리 분류 완료 - {이름}

### ISSUE THREAD

**Issue by J00NQ** [J00NQ/algorithm_YH#8]
## 실습 완료 내용

- [ ] Plant Pathology 2020 데이터 로드 & 탐색

- [ ] 이미지 전처리 (224×224 축소, 정규화)

- [ ] Dense 신경망 구축 (4-class 분류)

- [ ] 클래스 불균형 보정 (class_weight)

- [ ] 모델 훈련 (조기 종료 적용)

- [ ] 성능 평가 및 혼동행렬 분석

## 데이터셋 정보

- 원본 크기: 1365×1024 이미지

- 전처리: 224×224 축소 (메모리 효율)

- 클래스 분포:

  - Healthy: 516장

  - Multiple Diseases: 91장

  - Rust: 622장

  - Scab: 592장

## 모델 구축

- 입력층: 150,528 (224×224×3 이미지 flatten)

- 은닉층 1: Dense(512) + BatchNorm + Dropout(0.4) + ReLU

- 은닉층 2: Dense(256) + BatchNorm + Dropout(0.3) + ReLU

- 은닉층 3: Dense(128) + BatchNorm + Dropout(0.2) + ReLU

- 은닉층 4: Dense(64) + ReLU

- 출력층: Dense(4) + Softmax (다중분류)

## 성능 결과

- 훈련 정확도: 68.68%

- 검증 정확도: 44.11% (목표: 70% 이상)

- Log Loss: 1.2243(낮을수록 좋음)

- 훈련 에포크: 23개 (조기 종료 적용)

## 혼동행렬 분석 (4×4)

실제 \ 예측 | Healthy | Multiple | Rust | Scab
-- | -- | -- | -- | --
Healthy | - | 8 | 22 | 24
Multiple | - | - | 6 | -
Rust | 38 | - | - | 32
Scab | 23 | 8 | 32 | -

## 클래스별 정확도

- Healthy: 47.57% (재현율)

- Multiple Diseases: 22.22% (가장 어려운 클래스?)

- Rust: 41.60% 

- Scab: 47.06% 

## 관찰 기록

1. 이미지 축소 (1365×1024 → 224×224)가 성능에 미친 영향? 

2. 클래스 가중치 적용 전후 비교: 

3. 가장 많이 혼동되는 클래스 쌍: scep - rust

## 개선 시도 (선택)

- 시도 1: (예: 이미지 크기 조정) → 결과: ____%

- 시도 2: (예: Batch Normalization 제거) → 결과: ____%

- 가장 효과 있었던 변경: 

## 다음 단계 (내일)

CNN으로 같은 Plant Pathology 문제를 풀어서 Dense Network와 성능 비교 예정

## 커밋 및 링크

- Kaggle 노트북: (URL)

- GitHub: (링크)

**Comment by J00NQ** [comment]
<img width="678" height="791" alt="Image" src="https://github.com/user-attachments/assets/b434e9e5-c977-41bf-a1d8-f07d33c16b9b" />

**Comment by J00NQ** [comment]
<img width="1271" height="596" alt="Image" src="https://github.com/user-attachments/assets/c0629504-65ca-4f8e-835c-ccb195865b0a" />

**Comment by J00NQ** [comment]
<img width="644" height="648" alt="Image" src="https://github.com/user-attachments/assets/f1d0917b-9c3a-4c79-8e72-82d5eaccba20" />

**Comment by J00NQ** [comment]
<img width="919" height="773" alt="Image" src="https://github.com/user-attachments/assets/759425c8-f950-4890-9210-3594b86a01e7" />

**Comment by J00NQ** [comment]
<img width="1201" height="718" alt="Image" src="https://github.com/user-attachments/assets/eef4f891-530e-4a46-bb97-d380709c94cd" />

<img width="529" height="432" alt="Image" src="https://github.com/user-attachments/assets/5c4dd81d-06ea-49ba-bf16-5a2e770b7711" />

**Comment by J00NQ** [comment]
<img width="736" height="588" alt="Image" src="https://github.com/user-attachments/assets/c14cb337-7c94-4e7e-b9ad-05982cc744c9" />

<img width="464" height="217" alt="Image" src="https://github.com/user-attachments/assets/7538efbb-561d-4c8a-b4f8-cd293ef04081" />

<img width="296" height="106" alt="Image" src="https://github.com/user-attachments/assets/8d3010e3-1219-407a-b912-bd4af33c1c18" />

<img width="281" height="200" alt="Image" src="https://github.com/user-attachments/assets/24e6b414-519f-4386-bddf-aa51b791a511" />

**Comment by J00NQ** [comment]
시도 1: 데이터 증강(Multiple Diseases 불균형 해소) → 결과: 훈련 71.29%, 테스트 45.75%
